### PR TITLE
Storybook: Add storybook source display and ColorIndicator component

### PR DESCRIPTION
**Pull request by mkaz**
## Description

- Adds story for ColorIndicator component
- Change example color to WordPress blue (#0073AA) instead of red
- This PR expanded and now includes additions to storybook to display the source code on docs and main page.

## How has this been tested?

Run: `npm run design-system:dev`
Confirm ColorIndicator component shows properly

## Types of changes

Storybook. 

A webpack.config.js file was added to `packages/components/storybook/webpack.config.js` for a custom loader for loading story files source.

### CODE REVIEW

**Comment by mkaz** [comment]
The info works for the top level component, but the included components show as `[object Object]` in the source. The ButtonGroup for example:
```
ButtonGroup>
  <[object Object]
    isPrimary
    style={{
      margin: '0 4px'
    }}
  >
    Button 1
  </[object Object]>
  <[object Object]
    isPrimary
    style={{
      margin: '0 4px'
    }}
  >
    Button 2
  </[object Object]>
</ButtonGroup>
```

**Comment by mkaz** [comment]
Switched to using `storysource` and `source-loader` which adds a Story tab which shows source on main view, and code on Docs view. See screenshots below for examples.

Screenshot from 2019-10-13 12-51-47

Screenshot from 2019-10-13 12-51-56

Screenshot from 2019-10-13 12-52-18

**Comment by mkaz** [comment]
The new file at `packages/components/storybook/webpack.config.js` might be causing problems with the `./bin/update-readmes.js` script and the block-editor component, but I can't figure out why or how.

File: package.json [hunk 1]
```diff
@@ -74,8 +74,11 @@
 		"@octokit/rest": "16.26.0",
 		"@storybook/addon-a11y": "5.2.3",
 		"@storybook/addon-docs": "5.2.3",
+		"@storybook/addon-storysource": "5.2.3",
 		"@storybook/addon-viewport": "5.2.3",
+		"@storybook/addons": "5.2.3",
 		"@storybook/react": "5.2.3",
+		"@storybook/source-loader": "5.2.3",
```
Comment (epiqueras) [inline package.json]: Are you sure we need these two? The docs don't mention it.

File: packages/components/storybook/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module.exports = function( { config } ) {
+	config.module.rules.push( {
+		test: /stories\/index.jsx?$/,
```
Comment (epiqueras) [inline packages/components/storybook/webpack.config.js]: We need to match `/\/.+\.mdx$/` and `/\/stories\/.+\.js$/`.

**Review by epiqueras — CHANGES_REQUESTED** [review]
The dependency changes are making `@wordpress/docgen` throw on `Autogenerated selectors packages/block-editor/src/store/selectors.js`. Can we try to debug the root cause or test different versions?

File: package.json (hunk 1 above)
Comment (mkaz) [reply package.json]: Removed and cleaned up packages which fixed other issue. 
I was testing numerous different ways and I think the package-lock.json got some weird stuff in it

File: packages/components/storybook/webpack.config.js (hunk 2 above)
Comment (mkaz) [reply packages/components/storybook/webpack.config.js]: I improved the match, so its not just index file.

Testing out using the source-loader on `introduction.story.mdx` and it gave some weird results, I'm not sure if a different loader is needed for that format.

**Comment by mkaz** [comment]
Looks like mdx is not yet possible with `storysource` 
The releveant issue here: https://github.com/storybookjs/storybook/issues/8157

I think we can can move forward with this and follow up later for mdsx. The only mdx file now it the introduction post, which is just hello world, so not having source for that is fine.

**Comment by epiqueras** [comment]
Makes sense. I imagine they still have to refactor the loader to look inside `.mdx` files for code snippets.

File: packages/components/src/color-indicator/README.md [hunk 3]
```diff
@@ -6,6 +6,6 @@
 import { ColorIndicator } from '@wordpress/components';
 
 const MyColorIndicator = () => (
-	<ColorIndicator colorValue="#f00" />
+	<ColorIndicator colorValue="#0073AA" />
```
Comment (epiqueras) [inline packages/components/src/color-indicator/README.md]:
```suggestion
	<ColorIndicator colorValue="#0073aa" />
```

File: packages/components/src/color-indicator/stories/index.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+ */
+import ColorIndicator from '../';
+
+export default {
+	title: 'Color Indicator',
+	component: ColorIndicator,
+};
+
+export const _default = () => (
+	<ColorIndicator colorValue="#0073AA" />
```
Comment (epiqueras) [inline packages/components/src/color-indicator/stories/index.js]:
```suggestion
	<ColorIndicator colorValue="#0073aa" />
```

File: packages/components/storybook/addons.js [hunk 5]
```diff
@@ -3,3 +3,4 @@
  */
 import '@storybook/addon-a11y/register';
 import '@storybook/addon-viewport/register';
```
Comment (epiqueras) [inline packages/components/storybook/addons.js]:
```suggestion
import '@storybook/addon-storysource/register';
```

File: packages/components/storybook/addons.js [hunk 6]
```diff
@@ -3,3 +3,4 @@
...
+import '@storybook/addon-storysource/register';
```
Comment (epiqueras) [inline packages/components/storybook/addons.js]:
```suggestion
import '@storybook/addon-viewport/register';
```

File: packages/components/storybook/webpack.config.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+module.exports = function( { config } ) {
```
Comment (epiqueras) [inline packages/components/storybook/webpack.config.js]:
```suggestion
module.exports = ( { config } ) => {
```

**Review by epiqueras — APPROVED** [review]

**Pull request merged**